packages/flutter/lib/src/services/raw_keyboard.dart

```diff
@@ -499,12 +499,71 @@ class RawKeyUpEvent extends RawKeyEvent {
   }) : super(data: data, character: character);
 }
 
+/// An enum that describes how to handle a key event.
+enum KeyEventDisposition {
+  /// The key event has been handled, and the event should not be propagated to
+  /// other key event handlers.
+  handled,
+  /// The key event has not been handled, and the event should continue to be
+  /// propagated to other key event handlers, even non-Flutter ones.
+  ignored,
+  /// The key event has not been handled, but the key event should not be
+  /// propagated to other Flutter key event handlers, but should be propagated
+  /// to text fields, and non-Flutter key event handlers on the platform.
+  stopPropagation,
+}
+
+/// A class that represents a result returned from a [RawKeyEventHandler], which
+/// is invoked when a key event is received.
+class KeyEventResult {
```

Comment:
Why do we wrap the enum in a class?

Reply:
So that if we later want to add a payload to the result (perhaps the result of executing the action), we can do that without a breaking change.